lib/spack/llnl/util/filesystem.py

```diff
@@ -1180,6 +1187,16 @@ def remove_if_dead_link(path):
         os.unlink(path)
 
 
+def error_remove_readonly(func, path, exc):
+    excvalue = exc[1]
+    if is_windows and func in (os.rmdir, os.remove, os.unlink)\
+       and excvalue.errno == errno.EACCES:
+        # change the file to be readable,writable,executable: 0777
+        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)
+        # retry
+        func(path)
```

Comment:
I had to stare at this for a while to realize that this effectively preserves `ignore_errors=True` on Linux because this function doesn't raise any exceptions (or do anything) on Linux. I think it would be more straightforward to update how the `kwargs` are built in `remove_linked_tree` (i.e. on Linux `ignore_errors=True` and on Windows `ignore_errors=False` + `onerror=...`).